data_processors/pipeline/lambdas/orchestrator.py

```diff
@@ -345,63 +344,47 @@ def prepare_germline_jobs(this_batch: Batch, this_batch_run: BatchRun, this_sqr:
     job_list = []
     fastq_list_rows: List[dict] = libjson.loads(this_batch.context_data)
 
-    # get metadata for determining which sample need to be run through the germline workflow
-    metadata: dict = demux_metadata.handler({
-        'gdsVolume': this_sqr.gds_volume_name,
-        'gdsBasePath': this_sqr.gds_folder_path,
-        'gdsSamplesheet': this_sqr.sample_sheet_name,
-    }, None)
-
-    metadata_df = pd.DataFrame(metadata)
-    fastq_list_df = pd.DataFrame(fastq_list_rows)
-
-    # Add the sample_library_names attribute to fastq_list_df
-    fastq_list_df["sample_library_names"] = fastq_list_df['rgid'].apply(lambda x: x.rsplit('.', 1)[-1])
-
     # iterate through each sample group by rglb
-    for sample_library_name, sample_df in fastq_list_df.groupby("sample_library_names"):
+    for row in fastq_list_rows:
+        rgid = row['rgid']
 
-        # skip Undetermined samples
-        if sample_library_name.startswith("Undetermined"):
-            logger.warning(f"SKIP '{sample_library_name}' SAMPLE GERMLINE WORKFLOW LAUNCH.")
+        if row['read_1'].startswith("Undetermined"):
+            logger.info(f"SKIP GERMLINE workflow for {rgid}. Undetermined FASTQs.")
             continue
 
-        # skip sample start with NTC_
-        if sample_library_name.startswith("NTC_"):
-            logger.warning(f"SKIP NTC SAMPLE '{sample_library_name}' GERMLINE WORKFLOW LAUNCH.")
+        # Get the metadata for the library
+        # NOTE: this will use the library base ID (i.e. without topup/rerun extension), as the metadata is the same
+        rglb = row['rglb']
+        print(f"DEBUG {rglb}")
+        lib_metadata: LabMetadata = LabMetadata.objects.get(library_id=rglb)
```

Comment:
I reckon, here also, we are aiming not to depends on `sample_library_names` anymore (i.e. SampleSheet's Sample_ID style). 

However, with the current Lab Metadata, this may need to handle multiple metadata entries found for the said **_pure_** Library ID.  

As well as, Django `.objects.get()` will throw `DoesNotExit` exception before returning None. Hence, need handle it. Probably similar to BCL Convert situation -- we could have central method in services layer + model manager, perhaps...

Reply:
Agreed.
We should figure out if we support the legacy (multiple lab IDs) format or it we "clean" the data on import and rely on the DB consistency. 
One for discussion...

Reply:
Happy to defer code arrangement refactor into next round.